### PR TITLE
perf: Reduce hypersensitive onChange of watcher

### PR DESCRIPTION
**Pull request by 1natsu172**
refer #938 

Debounce the watcher's onChange.

### CODE REVIEW

File: packages/wxt/package.json [hunk 1]
```diff
@@ -81,8 +81,8 @@
     "prepack": "pnpm build"
   },
   "dependencies": {
-    "@types/chrome": "^0.0.269",
     "@aklinker1/rollup-plugin-visualizer": "5.12.0",
+    "@types/chrome": "^0.0.269",
```
Comment (1natsu172) [inline packages/wxt/package.json:85]: auto sorted by pnpm

File: packages/wxt/src/core/create-server.ts [hunk 2]
```diff
@@ -130,7 +131,9 @@ export async function createServer(
   });
 
   // Listen for file changes and reload different parts of the extension accordingly
-  const reloadOnChange = createFileReloader(server);
+  const reloadOnChange = debounce(createFileReloader(server), 3e2, {
+    leading: true,
+  });
```
Comment (1natsu172) [inline packages/wxt/src/core/create-server.ts:136]: I'm expecting unrealistic and excessive changes by machine. Not sure if the wait-time should be shorter or longer.... 🤔

Comment (pnd280) [reply packages/wxt/src/core/create-server.ts:136]: 300ms still causes the double reloads issue, I have to bump it to at least 3s

Comment (1natsu172) [reply packages/wxt/src/core/create-server.ts]: @pnd280 I turned off trailing, so 300ms might have been OK. Please check. If it's still NG, please send us a captured video of your screen.

**Comment by 1natsu172** [comment]
## NOTE: perfect-debounce behavior

<details><summary>Details</summary>
<p>

### `{ leading: true, trailing: true }`
image

### `{ leading: false, trailing: false }`
image


### `{ leading: true, trailing: false }`
image


### `{ leading: false, trailing: true }`
image

</p>
</details> 

`maxWait`
**There is no `maxWait`, so `leading` has priority. unlike lodash, even `{ leading: true, trailing: true }` does not fire a total of two times.**

File: packages/wxt/src/core/wxt.ts [hunk 3]
```diff
@@ -31,6 +32,8 @@ export async function registerWxt(
   // See https://github.com/wxt-dev/wxt/issues/873#issuecomment-2254555523
   process.env.NODE_ENV ??= command === 'serve' ? 'development' : 'production';
 
+  loadDotenv({ path: ['.env.wxt', '.env'], quiet: true });
```
Comment (1natsu172) [inline packages/wxt/src/core/wxt.ts:35]: Is this the right place to load? More top level? (cli.mjs?)

File: packages/wxt/src/types.ts [hunk 4]
```diff
@@ -904,6 +904,10 @@ export interface ConfigEnv {
    * details.
    */
   manifestVersion: 2 | 3;
+  /**
+   * WatchDebounce passed in from `.env.wxt` via the `WXT_WATCH_DEBOUNCE`. Which can suppress cases where dev server reloads are oversensitive.
+   */
+  watchDebounce: number;
```
Comment (1natsu172) [inline packages/wxt/src/types.ts:910]: Is this the right place to set in? Or should separate to another type and config property?

**Review by 1natsu172 — COMMENTED** [review]
@aklinker1 I implemented env variable loading and set the default value to a tentative 800ms (based on their logs). Please review when you have time!

**Review by aklinker1 — APPROVED** [review]
@1natsu172 sorry for the slow review on this! I made a couple of changes:

1. Move config from `ConfigEnv` to `ResolvedConfig` because environment variables isn't what the "env" is referencing in "ConfigEnv". That object is more for storing information about the current build setup than storing all environment variables
2. Updated .env loading behavior to match Vite's. See https://github.com/wxt-dev/wxt/blob/ef30d9aac1e7353a96ecf30c53df39b74a707516/packages/wxt/src/core/utils/env.ts
3. Moved loading of environment variables into `resolveConfig` so it refreshes the environment variables on save
4. Replaced `@dotenvx/dotenvx` with `dotenv` since we don't need the CLI, just the env loading behavior.

Otherwise it looks good! Thanks for fielding this one!

**Comment by 1natsu172** [comment]
@aklinker1 Oh you've already made a lot of changes and adjustments. Thanks!

**Comment by aklinker1** [comment]
@1natsu172 want me to revert any of that? I figured since it took me a while to review, I'd make the changes myself to spend things up

**Comment by 1natsu172** [comment]
@aklinker1 No, it's okay! I was concerned about the type of env file, timing of loading, etc., but everything is clear now 👍

**Pull request merged**